graphql/parser_test.go

```diff
@@ -248,3 +248,49 @@ fragment foo on Foo {
 		t.Error("expected unused fragment to fail", err)
 	}
 }
+
+func TestParseVariableDefinitions(t *testing.T) {
+	// Expect required variables to be provided.
+	_, err := Parse(`
+query Operation($x: int64!) {
+	field(x: $x)
+}	`, map[string]interface{}{})
+
+	if err == nil || err.Error() != "required variable not provided: $x" {
+		t.Error("expected unfulfilled required argument to fail, but got", err)
+	}
+}
+
+func TestParseRequiredVariableDefinitionWithDefaultValue(t *testing.T) {
+	// Expect required variables to be provided.
```

Comment:
should this comment say something like "expect required variables not to have default values"?